src/fileWriters/index.js

```diff
@@ -1,12 +1,56 @@
-import path from 'path';
+import extractZip from 'extract-zip';
+import chalk from 'chalk';
+import axios from 'axios';
+import fse from 'fs-extra';
+import rimraf from 'rimraf';
 
 import writeTemplateDirectoryFiles from './writeTemplateDirectoryFiles';
+import createTemplateFiles from './createTemplateFiles';
+import {
+  SOURCE_URL,
+  SOURCE_FILENAME,
+  TEMP_DIR,
+  TEMPLATES_DIR,
+} from '../constants';
 
-const writeBaseTemplates = async ({ destination, templateValues }) =>
+export const writeBaseTemplates = async ({ destination, templateValues }) =>
   writeTemplateDirectoryFiles({
     destination,
     templateValues,
-    location: path.join(__dirname, '../templates'),
+    location: TEMPLATES_DIR,
   });
 
-export default writeBaseTemplates;
+const extractSource = async (source) => {
+  extractZip(source, { dir: TEMP_DIR }, async (error) => {
+    if (error) {
+      console.log(chalk.bold.redBright(error));
+      return;
+    }
+    await createTemplateFiles();
+  });
+};
+
+export const getCookieCutterSource = async () => {
+  try {
+    console.log(chalk.bold.blueBright(`🐶  Fetching source from ${SOURCE_URL}...`));
+    console.log();
+
+    const response = await axios({
+      url: SOURCE_URL,
+      method: 'get',
+      responseType: 'arraybuffer',
+    });
+
+    if (fse.existsSync(TEMP_DIR)) {
```

Comment:
This feels dangerous to me - if I fat-finger a directory name this is going to blow away that directory without even prompting me?

Reply:
This is just clearing out the `tmp` directory that may or may not already exist. The `TEMP_DIR` constant is defined in `constants.js`; the user isn't inputting the directory name here. Is your question more about if someone decides to change the `tmp` directory name in `constants.js`?

Reply:
no, I misread this - totally fine.